### PR TITLE
feat(rest): list all transitive releases for a specific project

### PR DESCRIPTION
**Pull request by maierthomas**
New optional request parameter for project controller (transitive)

e.g. 
```
without request parameter / or false:
/resource/api/projects/123456/releases
/resource/api/projects/123456/releases/ecc
/resource/api/projects/123456/releases?transitive=false
/resource/api/projects/123456/releases?transitive=false/ecc
```
```
with request parameter: transitive
/resource/api/projects/123456/releases?transitive=true
/resource/api/projects/123456/releases?transitive=true/ecc
```

closes sw360/sw360portal#733

### CODE REVIEW

**Comment by adrian-evo** [comment]
This works:
releases -> display linked releases
releases/ecc -> display linked releases and eccStatus

This does not work:
releases?transitive=false/ecc -> display linked releases but not eccStatus
releases?transitive=true -> display only linked releases but not subproject releases

**Comment by maierthomas** [comment]
Sorry my fault. please try the following endpoints.
The request parameter 

> transitive

 should be at the end of the endpoint.

```
without request parameter / or false:
/resource/api/projects/123456/releases
/resource/api/projects/123456/releases/ecc
/resource/api/projects/123456/releases?transitive=false
/resource/api/projects/123456/releases/ecc?transitive=false

with request parameter: transitive
/resource/api/projects/123456/releases?transitive=true
/resource/api/projects/123456/releases/ecc?transitive=true

```

**Comment by adrian-evo** [comment]
Releases from linked project are not displayed, i.e. transitive=true has no effect.

p1

p3

p2

**Comment by maierthomas** [comment]
Please check and verify your deployment again.
I can see the linked releases ob sub projects.

releasesview

**Comment by adrian-evo** [comment]
OK, so it works on second deployment.

File: rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/project/Sw360ProjectService.java [hunk 1]
```diff
@@ -121,6 +124,21 @@ public void deleteAllProjects(User sw360User) {
...
+            ProjectService.Iface sw360ProjectClient = getThriftProjectClient();
+            if (Boolean.parseBoolean(transitive)) {
+                List<ReleaseClearingStatusData> releaseClearingStatusData = sw360ProjectClient.getReleaseClearingStatuses(projectId, sw360User);
+                return releaseClearingStatusData.stream().map(r -> r.release.getId()).collect(Collectors.toSet());
+            } else {
+                final Project project = getProjectForUserById(projectId, sw360User);
+                return project.getReleaseIdToUsage().keySet();
+            }
+        } catch (TException e) {
+            throw new RuntimeException(e);
```
Comment (alexbrdn) [inline rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/project/Sw360ProjectService.java]: I'm not a fan of throwing a generic exception. I think `getReleaseIds()` should declare `throws TException` and `getProjectReleases()` should catch the exception and return a proper `ResponseEntity` with HTTP 500.

Comment (maierthomas) [reply rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/project/Sw360ProjectService.java]: fixed exception handling for controller and service classes

**Pull request merged**